src/Uno.UI/UI/Xaml/Documents/TextElement.cs

```diff
@@ -167,22 +167,27 @@ protected virtual void OnFontSizeChanged()
 
 		public Brush Foreground
 		{
-			get => GetForegroundValue();
+			get => (Brush)GetValue(ForegroundProperty);
 			set
 			{
 				if (value != null && !(value is SolidColorBrush))
 				{
 					throw new InvalidOperationException("Specified brush is not a SolidColorBrush");
 				}
 
-				SetForegroundValue(value);
+				SetValue(ForegroundProperty, value);
 			}
 		}
 
-		[GeneratedDependencyProperty(Options = FrameworkPropertyMetadataOptions.Inherits, ChangedCallback = true, ChangedCallbackName = nameof(OnForegroundChanged))]
-		public static DependencyProperty ForegroundProperty { get; } = CreateForegroundProperty();
-
-		private static Brush GetForegroundDefaultValue() => SolidColorBrushHelper.Black;
+		public static DependencyProperty ForegroundProperty { get; } = DependencyProperty.Register(
+			nameof(Foreground),
+			typeof(Brush),
+			typeof(TextElement),
+			new FrameworkPropertyMetadata(
+				defaultValue: SolidColorBrushHelper.Black,
+				options: FrameworkPropertyMetadataOptions.Inherits,
+				propertyChangedCallback: (instance, args) => ((TextElement)instance).OnForegroundChanged()
+		));
```

Comment:
What caused the generator to be unuseable here?

Reply:
Due to the way default value work (it's no longer stored, but calculated), we no longer have the backing field update when theme changes. This is also closer to WinUI where property changed isn't fired when theme changes.